homeassistant/components/proximity/coordinator.py

```diff
@@ -164,16 +164,21 @@ def _calc_distance_to_zone(
             )
             return None
 
-        distance_to_zone = distance(
+        distance_to_centre = distance(
             zone.attributes[ATTR_LATITUDE],
             zone.attributes[ATTR_LONGITUDE],
             latitude,
             longitude,
         )
 
         # it is ensured, that distance can't be None, since zones must have lat/lon coordinates
-        assert distance_to_zone is not None
-        return round(distance_to_zone)
+        assert distance_to_centre is not None
+
+        zone_radius: float = zone.attributes["radius"]
+        distance_to_edge: int = round(distance_to_centre - zone_radius)
+        # return 0 if distance to edge is negative as that means the device is located inside the zone
+        # but the device's current zone is a different one than the zone in this distance calculation
```

Comment:
```suggestion
        # return 0 when we've arrived the zone
```

it might be possible, that the device is also within another zone, but this doesn't matter at this point, since this calculation is for the current specific zone - so if it reaches a distance of 0, it is in this specific zone.

So technically this calculation is correct, but the comments are misleading

Reply:
I thought it would be good to point out that this only applies when the device is in a different zone. That's because further up in the same method, we check for the device's zone and return 0 if it matches, so the max only applies if we have overlapping zones. I'm fine with your suggestion though, since it's correct regardless of the previous checks.